### ISSUE TITLE
`termux-call-log` doesn't work

### ISSUE THREAD

**Issue by aryadovoy** [termux/termux-api#434]
No output.
`termux-sms-list` is fine, Termux:API has access to Call logs, Phone, SMS.

Google Pixel 4a 5G, Android 11.

**Comment by Grimler91** [comment]
I can reproduce the issue. Logcat shows

```
07-24 23:21:08.732 30341  3161 E termux-api: Error in ResultReturner
07-24 23:21:08.732 30341  3161 E termux-api: java.lang.IllegalArgumentException: Invalid token LIMIT
07-24 23:21:08.732 30341  3161 E termux-api:    at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:172)
07-24 23:21:08.732 30341  3161 E termux-api:    at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:142)
07-24 23:21:08.732 30341  3161 E termux-api:    at android.content.ContentProviderProxy.query(ContentProviderNative.java:472)
07-24 23:21:08.732 30341  3161 E termux-api:    at android.content.ContentResolver.query(ContentResolver.java:1183)
07-24 23:21:08.732 30341  3161 E termux-api:    at android.content.ContentResolver.query(ContentResolver.java:1115)
07-24 23:21:08.732 30341  3161 E termux-api:    at android.content.ContentResolver.query(ContentResolver.java:1071)
07-24 23:21:08.732 30341  3161 E termux-api:    at com.termux.api.CallLogAPI.getCallLogs(CallLogAPI.java:39)
07-24 23:21:08.732 30341  3161 E termux-api:    at com.termux.api.CallLogAPI.access$000(CallLogAPI.java:21)
07-24 23:21:08.732 30341  3161 E termux-api:    at com.termux.api.CallLogAPI$1.writeJson(CallLogAPI.java:29)
07-24 23:21:08.732 30341  3161 E termux-api:    at com.termux.api.util.ResultReturner$ResultJsonWriter.writeResult(ResultReturner.java:88)
07-24 23:21:08.732 30341  3161 E termux-api:    at com.termux.api.util.ResultReturner.lambda$returnData$0(ResultReturner.java:133)
07-24 23:21:08.732 30341  3161 E termux-api:    at com.termux.api.util.-$$Lambda$ResultReturner$oCU8JqHW5HLMrNZa40ThjAWcwzo.run(Unknown Source:8)
07-24 23:21:08.732 30341  3161 E termux-api:    at java.lang.Thread.run(Thread.java:923)
```

**Comment by Usainlit** [comment]
> I can reproduce the issue. Logcat shows
> 
> ```
> 07-24 23:21:08.732 30341  3161 E termux-api: Error in ResultReturner
> 07-24 23:21:08.732 30341  3161 E termux-api: java.lang.IllegalArgumentException: Invalid token LIMIT
> 07-24 23:21:08.732 30341  3161 E termux-api:    at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:172)
> 07-24 23:21:08.732 30341  3161 E termux-api:    at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:142)
> 07-24 23:21:08.732 30341  3161 E termux-api:    at android.content.ContentProviderProxy.query(ContentProviderNative.java:472)
> 07-24 23:21:08.732 30341  3161 E termux-api:    at android.content.ContentResolver.query(ContentResolver.java:1183)
> 07-24 23:21:08.732 30341  3161 E termux-api:    at android.content.ContentResolver.query(ContentResolver.java:1115)
> 07-24 23:21:08.732 30341  3161 E termux-api:    at android.content.ContentResolver.query(ContentResolver.java:1071)
> 07-24 23:21:08.732 30341  3161 E termux-api:    at com.termux.api.CallLogAPI.getCallLogs(CallLogAPI.java:39)
> 07-24 23:21:08.732 30341  3161 E termux-api:    at com.termux.api.CallLogAPI.access$000(CallLogAPI.java:21)
> 07-24 23:21:08.732 30341  3161 E termux-api:    at com.termux.api.CallLogAPI$1.writeJson(CallLogAPI.java:29)
> 07-24 23:21:08.732 30341  3161 E termux-api:    at com.termux.api.util.ResultReturner$ResultJsonWriter.writeResult(ResultReturner.java:88)
> 07-24 23:21:08.732 30341  3161 E termux-api:    at com.termux.api.util.ResultReturner.lambda$returnData$0(ResultReturner.java:133)
> 07-24 23:21:08.732 30341  3161 E termux-api:    at com.termux.api.util.-$$Lambda$ResultReturner$oCU8JqHW5HLMrNZa40ThjAWcwzo.run(Unknown Source:8)
> 07-24 23:21:08.732 30341  3161 E termux-api:    at java.lang.Thread.run(Thread.java:923)
> ```

**Comment by navid-zamani** [comment]
Doesn’t work on Andoid 12 (EMUI 12) either. To be precise: It always outputs an empty list, aka nothing. Logcat shows the same error here too.

**Comment by Shivam623** [comment]
any updates on this? call logs not working for me as well. :(

**Comment by navid-zamani** [comment]
It’s getting critical, as I’m using this function, to back up my call logs, since I stopped being a ”cloud“-used data prostitute some time ago. So I really need it.

It seems to be a simple matter of adapting it to the new underlying parameters. But I avoid learning the gslopping i sanity that is Android internals, so I don’t know how to fix this in practice.

Can we do anything to help?

**Comment by agnostic-apollo** [comment]
Either wait for F-Droid release whenever that happens or (move and) use github builds.

https://github.com/termux/termux-api/actions/runs/3115767801

**Comment by licaon-kter** [comment]
Would be nice if this was actually tagged and released so F-Droid can pick it. ;)

**Comment by happysmash27** [comment]
> Either wait for F-Droid release whenever that happens or (move and) use github builds.
> 
> https://github.com/termux/termux-api/actions/runs/3115767801

@agnostic-apollo I desperately and urgently need to use this feature or any other method to automatically play audio on my computer that does not end when the call ends, so that I can wake up when an important call comes on my phone, ASAP, but the build fails to install due to: 

`apk: Failure [INSTALL_FAILED_SHARED_USER_INCOMPATIBLE: Reconciliation failed...: Reconcile failed: Package com.termux.api has no signatures that match those in shared user com.termux; ignoring!]`

Alternatively, is this bug fixed on earlier versions on F-Droid, or is there a way to easily migrate the files from the existing Termux installation to a self-signed one on a phone (OnePlus Nord N100) which nobody has come up with a good root or custom bootloader solution for yet?

**Comment by Lucas-Sichardt** [comment]
Hello together, 

I just ran into the same issue I think. Is this still not solved? I have v0.118.0 and I don't seem to find a newer version even on GitHub. Is there any?
